pkg/client/keycloak/config.go

```diff
@@ -133,5 +149,82 @@ func (kc *KeycloakConfig) ReadFiles() error {
 	kc.OSDClusterIDPRealm.setDefaultURIs(kc.BaseURL)
 	kc.RedhatSSORealm.setDefaultURIs(kc.SsoBaseUrl)
 
+	// Read the additional endpoints file. This will add additional SSO endpoints which shall be used as valid issuers
+	// for tokens, i.e. sso.stage.redhat.com.
+	if kc.AdditionalSSOEndpoints.EnableAdditionalSSOIssuers {
+		err = readAdditionalEndpointsFile(kc.AdditionalSSOEndpoints.AdditionalSSOIssuersFile, kc.AdditionalSSOEndpoints)
+		if err != nil {
+			if os.IsNotExist(err) {
+				glog.V(10).Infof("Specified additional SSO endpoints file %q does not exist. "+
+					"Proceeding as if no additional SSO endpoints list was provided", kc.AdditionalSSOEndpoints.AdditionalSSOIssuersFile)
+			} else {
+				return err
+			}
+		}
+		if err := kc.AdditionalSSOEndpoints.resolveURIs(); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+const (
+	openidConfigurationPath = "/.well-known/openid-configuration"
+)
+
+type openIdConfiguration struct {
+	JwksURI string `json:"jwks_uri"`
+}
+
+// setJWKSURIs will set the jwks URIs by taking the issuer URI and fetching the openid-configuration, setting the
+// jwks URI dynamically
+func (a *AdditionalSSOEndpoints) resolveURIs() error {
+	client := http.Client{Timeout: time.Minute}
+	jwksURIs := make([]string, 0, len(a.IssuerURIs))
+	for _, issuerURI := range a.IssuerURIs {
+		cfg, err := getOpenIDConfiguration(client, issuerURI)
+		if err != nil {
+			return errors.Wrapf(err, "retrieving open-id configuration for %q", issuerURI)
+		}
+		if cfg.JwksURI == "" {
+			return errors.Errorf("no jwks URI found within open-id configuration for %q", issuerURI)
+		}
+		jwksURIs = append(jwksURIs, cfg.JwksURI)
+	}
+	a.JWKSURIs = jwksURIs
+	return nil
+}
+
+func getOpenIDConfiguration(c http.Client, baseURL string) (*openIdConfiguration, error) {
+	url := baseURL + openidConfigurationPath
```

Comment:
nitpick: To be on the safe side we should trim right `/`. Sometimes copy paste slips and we copy URLs with slash at the end.
```suggestion
	url := strings.TrimRight(baseURL, "/") + openidConfigurationPath
```